prow/spyglass/spyglass_test.go

```diff
@@ -445,6 +446,147 @@ func TestJobPath(t *testing.T) {
 	}
 }
 
+func TestProwJobPath(t *testing.T) {
+	kc := fkc{
+		prowapi.ProwJob{
+			ObjectMeta: metav1.ObjectMeta{Name: "flying-whales-1"},
+			Spec: prowapi.ProwJobSpec{
+				Type: prowapi.PeriodicJob,
+				Job:  "example-periodic-job",
+				DecorationConfig: &prowapi.DecorationConfig{
+					GCSConfiguration: &prowapi.GCSConfiguration{
+						Bucket: "chum-bucket",
+					},
+				},
+			},
+			Status: prowapi.ProwJobStatus{
+				PodName: "flying-whales",
+				BuildID: "1111",
+			},
+		},
+		prowapi.ProwJob{
+			ObjectMeta: metav1.ObjectMeta{Name: "flying-whales-2"},
+			Spec: prowapi.ProwJobSpec{
+				Type: prowapi.PresubmitJob,
+				Job:  "example-presubmit-job",
+				DecorationConfig: &prowapi.DecorationConfig{
+					GCSConfiguration: &prowapi.GCSConfiguration{
+						Bucket: "chum-bucket",
+					},
+				},
+			},
+			Status: prowapi.ProwJobStatus{
+				PodName: "flying-whales",
+				BuildID: "2222",
+			},
+		},
+		prowapi.ProwJob{
+			ObjectMeta: metav1.ObjectMeta{Name: "flying-whales-3"},
+			Spec: prowapi.ProwJobSpec{
+				Type: prowapi.PresubmitJob,
+				Job:  "undecorated-job",
+			},
+			Status: prowapi.ProwJobStatus{
+				PodName: "flying-whales",
+				BuildID: "1",
+			},
+		},
```

Comment:
nit: I don't think we need test cases for all the job types and undecorated jobs since neither of those things should be related to getting the ProwJob name.